### PR TITLE
Contributing document describing the basics of how to contribute validator rules for amp components.

### PR DESCRIPTION
**Pull request by Gregable**

### CODE REVIEW

File: contributing/component-validator-rules.md [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+  }
+}
+```
+
+## Final Note on Rules
+
+This document attempts to summarize some of the more commonly used rules for
+creating validator extended components. More complex rules are possible and new rule
+types can even be added as needed. If your goals are not met by the rules in
+this document, don't hesitate to contact the AMP developers and ask for
```
Comment (mrjoro) [inline contributing/component-validator-rules.md]: perhaps link "don't hesitate to contact" to the discussion channels section of CONTRIBUTING:

https://github.com/ampproject/amphtml/blob/503f8766bbb097ac970c90b3aea21b6e8c98ea4d/CONTRIBUTING.md#discussion-channels

**Comment by aghassemi** [comment]
@mrjoro @Gregable yes, let's also link to it from https://docs.google.com/document/d/19o7eDta6oqPGF4RQ17LvZ9CHVQN53whN-mCIeIMM8Qk/edit under the "Allowing Proper Validations" section. (this is publicly available Google doc for building an AMP component)

File: contributing/component-validator-rules.md [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+<!-- Display the cat named 'oscar' -->
+<amp-cat data-selected-cat="oscar" width=50 height=50></amp-cat>
+
+<!-- Display a random cat -->
+<amp-cat width=50 height=50></amp-cat>
+```
+
+Your first step will be writing the extended component JavaScript code. The
+code will be placed in the amphtml src tree at the location of
+`amphtml/extensions/amp-cat/0.1/`. However, this document only descriptes how to
```
Comment (Gregable) [inline contributing/component-validator-rules.md]: done

File: contributing/component-validator-rules.md [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+
+> `<script async custom-element='amp-cat'`
+> `        src='https://cdn.ampproject.org/v0/amp-cat-0.1.js'></script>`
+
+```
+  html_format: AMP
+```
+
+
+and that this tag
```
Comment (Gregable) [inline contributing/component-validator-rules.md]: done

File: contributing/component-validator-rules.md [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+```
+These fields define a list of all allowed version numbers. Currently, almost all
+extended components are at version `0.1`, and we also allow `latest` to be specified.
+
+The combination of the `allowed_versions` and `name` fields of the
+`extension_spec` define the allowed values of the `src` attribute in the
+script tag, for example `src=https://cdn.ampproject.org/v0/amp-cat-0.1.js`.
+
+```
+  }
```
Comment (Gregable) [inline contributing/component-validator-rules.md]: The space is intentional, but it's clearer moved into the previous code block.

**Comment by Gregable** [comment]
@aghassemi Will do once committed, thanks for the reference.

**Comment by honeybadgerdontcare** [comment]
/sub to be notified when this PR gets merged

**Pull request merged**